packages/model-viewer/src/three-components/gltf-instance/VariantMaterialLoaderPlugin.ts

```diff
@@ -0,0 +1,275 @@
+/**
+ * Materials variants extension
+ *
+ * Specification:
+ * https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_variants
+ */
+
+import {Material, Mesh, Object3D} from 'three';
+import {GLTF, GLTFLoaderPlugin, GLTFParser} from 'three/examples/jsm/loaders/GLTFLoader.js';
+
+/**
+ * KHR_materials_variants specification allows duplicated variant names
+ * but it makes handling the extension complex.
+ * We ensure tha names and make it easier.
+ * If you want to export the extension with the original names
+ * you are recommended to write GLTFExporter plugin to restore the names.
+ *
+ * @param variantNames {Array<string>}
+ * @return {Array<string>}
+ */
+const ensureUniqueNames = (variantNames: string[]) => {
+  const uniqueNames = [];
+  const knownNames: {[key: string]: boolean} = {};

```

Comment:
Here's another place where `new Map<string, boolean>();` may be preferable.

Reply:
Done